### PR TITLE
Add ability to apply patches to any branch

### PR DESCRIPTION
**Pull request by rojvv**

### CODE REVIEW

File: deno.lock [hunk 1]
```diff
@@ -476,63 +279,36 @@
...
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/markup.d.ts": "c2eb6a0fe8f5e15d903bfa33f23f36ae5c9a77accc5d5656a80fe26ae9793057",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/menu-button.d.ts": "096e03dadaf3af415c48c14cb91380769df01dcccbb0931575bf5f15a5df5cd7",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/message.d.ts": "fb395a5ec2b130269d680788b098f32fec4ee2eeb3fa4aeead14fdb8b8095d03",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/passport.d.ts": "e3fb63aec96510bcc317ef48fd25b435444b8f407502d7568c00fce15f2958fd",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/payment.d.ts": "410c384a26d24252219713adeb0c360d5d038102ab9d1a6e9aad20b603830d0c",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/proxied.d.ts": "f633a4e6f59f34362d5dcd94872a661f0aa46e695ba0dd2445e844b5ef03fe19",
+    "https://esm.sh/v110/@grammyjs/types@2.12.1/update.d.ts": "7c4fb6534a1b401ab300f7edf49e2fe75293153fc44af14382eea0c0c02033c2",
+    "https://esm.sh/v110/@octokit/webhooks-methods@3.0.2/deno/webhooks-methods.js": "2169aef27a5052b2447bfe05132a4fccdaf044b439f0085b26a4f18680ef1294",
```
Comment (KnorpelSenf) [inline deno.lock]:
```suggestion
    "https://esm.sh/v110/@octokit/webhooks-methods@3.0.2/deno/webhooks-methods.js": "2169aef27a5052b2447bff05132a4fccdaf044b439f0085b26a4f18680ef1294",
```

**Comment by KnorpelSenf** [comment]
Joking

**Comment by rojvv** [comment]
LMAO

File: main.ts [hunk 2]
```diff
@@ -51,22 +68,31 @@ bot.chatType(["group", "supergroup"]).filter((ctx) =>
...
-  } else {
-    console.log(res.status);
-    await ctx.reply("Failed to submit the patch.");
+  } catch (err) {
+    await ctx.reply(err instanceof Error ? err.message : String(err));
+  }
+});
+
+bot.command("apply", async (ctx) => {
+  const branch = ctx.msg.text.split(/\s/)[1];
```
Comment (KnorpelSenf) [inline main.ts]: `ctx.match`?

Comment (rojvv) [reply main.ts]: Wouldn’t that need a `bot.on(`?

File: main.ts [hunk 3]
```diff
@@ -51,22 +68,31 @@ bot.chatType(["group", "supergroup"]).filter((ctx) =>
...
+        await ctx.reply("Could not resolve file path.");
+        return;
+      }
+      try {
+        await applyFile(branch, file_path);
+        await ctx.reply("Patch submitted.");
+      } catch (err) {
+        await ctx.reply(err instanceof Error ? err.message : String(err));
+      }
+    }
```
Comment (KnorpelSenf) [inline main.ts]: Probably mismatching brackets but
```suggestion
    } else {
      await ctx.reply("Please reply to a patch file with this command")
    }
  } else {
    await ctx.reply("Please provide a branch name in the format /apply <branch>.")
```

File: main.ts (hunk 2 above)
Comment (KnorpelSenf) [reply main.ts]: https://grammy.dev/guide/commands.html#arguments

Comment (rojvv) [reply main.ts]: Oy. Firmware upgrade available.

File: main.ts (hunk 3 above)
Comment (KnorpelSenf) [reply main.ts]: What about the branch name?

**Review by KnorpelSenf — APPROVED** [review]

**Review by KnorpelSenf — APPROVED** [review]
much better

**Pull request merged**